op-challenger/game/generic/player.go

```diff
@@ -135,7 +135,10 @@ func (g *GamePlayer) ProgressGame(ctx context.Context) gameTypes.GameStatus {
 		return g.status
 	}
 	g.logger.Trace("Checking if actions are required")
-	if err := g.actor.Act(ctx); err != nil {
+	if err := g.actor.Act(ctx); errors.Is(err, client.ErrNotInSync) {
+		g.logger.Warn("Local node not sufficiently up to date to act on game", "err", err)
```

Comment:
```suggestion
		g.logger.Error("Local node not sufficiently up to date to act on game", "err", err)
```

This shouldn't happen given the `ValidateNodeSynced` precheck. Suggests either a bug or a bad supernode RPC.

Reply:
Yeah I'm tossing up whether we should just skip the pre-validate step for the Supernode since we can handle it here. Otherwise the sync validator would just make this same `superroot_atTimestamp` request and check only the CurrentL1 value, throwing away the rest of the data.

I'm also thinking a bit more about how to better handle load balanced nodes where we might get one in sync response and then a not in sync response from another node. It would be nice to handle that nicely at least if the nodes are both healthy but may have a small lag in the latest update. That way you can have a proxy with pretty simple health checks that works, without needing to have the full consensus aware handling that we do for ELs.

Reply:
I'm not sure whether load balancing is needed (it's not expensive to load a super root is it?). But improving robustness against out of synced nodes would be nice indeed. The CurrentL1 is the only constraint to do this since you can check every node until there's one that's synced past the CurrentL1.

Reply:
Load balancing isn't required for performance, but is very useful for reliability. A load balancer could handle some nodes being down easily and with even very basic health checks could remove a node that's too far out of sync from active service. Just right now the challenger requires consistency across multiple calls which doesn't work with simple load balancers.  Fixing that single point of failure is a very common request and we've had quite a few cases where the challenger did the wrong thing because it was pointed at a load balancer and got inconsistent responses.